frontend/integration-tests/tests/storage.scenario.ts

```diff
@@ -0,0 +1,55 @@
+import { browser, ExpectedConditions as until, by, element } from 'protractor';
+import { appHost, testName, checkLogs, checkErrors } from '../protractor.conf';
+import * as crudView from '../views/crud.view';
+import { execSync } from 'child_process';
+import { OrderedMap } from 'immutable';
+import * as yamlView from '../views/yaml.view';
+
+describe('Add storage is applicable for all workloads', () => {
+
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  const k8sObjs = OrderedMap<string, {kind: string, namespaced?: boolean}>()
+    .set('replicationcontrollers', {kind: 'ReplicationController'})
+    .set('daemonsets', {kind: 'DaemonSet'})
+    .set('deployments', {kind: 'Deployment'})
+    .set('replicasets', {kind: 'ReplicaSet'})
+    .set('statefulsets', {kind: 'StatefulSet'});
+
+  const openshiftObjs = OrderedMap<string, {kind: string, namespaced?: boolean}>()
+    .set('deploymentconfigs', {kind: 'DeploymentConfig'});
+  const resourceObjs = browser.params.openshift === 'true' ? k8sObjs.merge(openshiftObjs) : k8sObjs;
+
+  resourceObjs.forEach(({kind}, resource) => {
+    describe(kind, () => {
+      it('create a new resource', async() => {
+        await browser.get(`${appHost}/k8s/ns/${testName}/${resource}`);
+        await crudView.isLoaded();
+        const createDropdownIsPresent = await crudView.createItemButton.isPresent();
+        if (createDropdownIsPresent) {
+          await crudView.createItemButton.click();
+          await crudView.createYAMLLink.click();
+        } else {
+          await crudView.createYAMLButton. click();
+        }
+        browser.wait(until.and(crudView.untilNoLoadersPresent, until.presenceOf(element(by.cssContainingText('h1', 'Create')))));
+        await yamlView.saveButton.click();
+        expect(crudView.errorMessage.isPresent()).toBe(false);
+      });
+      it('check add storage item in action list', async() => {
+        await browser.wait(until.presenceOf(crudView.actionsDropdown));
+        await crudView.actionsDropdown.click();
+        const item = element(by.cssContainingText('.dropdown li a', 'Add Storage'));
```

Comment:
Move the CSS selectors into `crudView`

Reply:
Thanks for your suggestion, will update soon.